### ISSUE TITLE
Loading a large amount of images crashes Godot

### ISSUE THREAD

**Issue by metanoder** [godotengine/godot#68822]
### Godot version

4.0 Beta 5

### System information

M1 MacBook Pro Arm 32gb

### Godot currently can't load large volumes of images all at once.  

I have a game on Godot 3.5 and use the plug in dialogic for my dialog.  They are developing a conversation feature that will port my game over to the 4.0 branch.  But after trying it, it crashes my editor as I have roughly 100 images.  

Even when I copy them over without converter, it crashes Godot.  
I found that it can only take between roughly 20-50 images at a time.  

Not ideal to not be able to upload a reasonable amount of images at a time.  

### Steps to reproduce

after starting up game engine, grab 20 - 100 images, and copy and paste into game folder in window res://.

Game Engine Crashes...

### Minimal reproduction project

N/A

**Comment by Calinou** [comment]
@metanoder Please upload a [minimal reproduction project](https://github.com/godotengine/godot/blob/master/CONTRIBUTING.md#provide-a-simple-example-project) to make this easier to troubleshoot.

How large are the images in total? What's their size in pixels on average? Any images that are really small or really large?

**Comment by metanoder** [comment]
I isolated problem to a single SVG.

Though, I cannot replicate crashes consistently...  ( Once was able to add files without problem.  )
I've tried a number of content creation software to create SVGs (CCS = [Vectornator, Affinity Designer, and Inkscape])

my process for svg...
I copied file image from apple os by hitting command "I" on file folder.  
Copy file icon from info tab popup.
opened CCS and paste.
Add star vector from CCS toolkit
export svg
import to godot resources folder
game crashes on open.

Project file zipped attached with svg from inkscape, but all svgs I've created with this crashed....

[svgBug.zip](https://github.com/godotengine/godot/files/10044437/svgBug.zip)

**Comment by akien-mga** [comment]
It doesn't crash for me, but it fails importing the SVG indeed and enters a loop of attempting to reimport and failing:
```
ERROR: ImageLoaderSVG can't create image.
   at: create_image_from_string (modules/svg/image_loader_svg.cpp:112)
ERROR: Condition "p_image->is_empty()" is true. Returning: FAILED
   at: load_image (modules/svg/image_loader_svg.cpp:154)
ERROR: Error loading image: res://testingSVG_fromInkscape_0.svg
   at: load_image (core/io/image_loader.cpp:101)
ERROR: Error importing 'res://testingSVG_fromInkscape_0.svg'.
   at: _reimport_file (editor/editor_file_system.cpp:1909)
ERROR: ImageLoaderSVG can't create image.
   at: create_image_from_string (modules/svg/image_loader_svg.cpp:112)
ERROR: Condition "p_image->is_empty()" is true. Returning: FAILED
   at: load_image (modules/svg/image_loader_svg.cpp:154)
ERROR: Error loading image: res://testingSVG_fromInkscape_0.svg
   at: load_image (core/io/image_loader.cpp:101)
ERROR: Error importing 'res://testingSVG_fromInkscape_0.svg'.
   at: _reimport_file (editor/editor_file_system.cpp:1909)
```

The reimport loop issue is tracked in this issue: #62271.

But this issue here can focus on the SVG import failure.

**Comment by akien-mga** [comment]
So the macOS icon in that SVG is a PNG, so it seems we fail loading a SVG with a bundled PNG.
ThorVG has a PNG loader so it's probably a misuse of the API on our end.

**Comment by akien-mga** [comment]
Seems like the upstream test project `svg2png` can't process that file either and it crashes, so I reported it there: https://github.com/Samsung/thorvg/issues/1278

**Comment by mgrudzinska** [comment]
@metanoder @akien-mga I hope the problem disappears after https://github.com/Samsung/thorvg/pull/1279
also please check the https://github.com/Samsung/thorvg/issues/1278#issuecomment-1321227915

**Comment by akien-mga** [comment]
Thanks @mgrudzinska!

If I understand correctly, aside from that SVG being a pathological case, the problem in Godot might be that since the canvas area is empty (the actual PNG icon + vector graphics star icon being outside the canvas), we fail drawing because there's nothing to draw?

The failure in Godot happens here:

https://github.com/godotengine/godot/blob/a8a88194a5cf5849d421aaec05beddbc437ebbcd/modules/svg/image_loader_svg.cpp#L109-L113

I assume this is then an expected error condition for `SwCanvas::draw`? https://samsung.github.io/thorvg/html/classtvg_1_1Canvas.html#aabf9e5a14da3b4bfd5883c2b4459abd0

I can try to improve the error handling here. Maybe we can also provide users with useful feedback that the canvas is empty before attempting to draw - is there any easy way to check that? (I haven't spent much time with thorvg myself yet, the implementation was done by @fire.)

**Comment by metanoder** [comment]
@akien-mga 

Yep, the file I uploaded is definitely "off" due to how I made it...   that was my mistake.  

However, I tried this in multiple instances without the canvas oversized / non-phycological and still crashes if used with **MacOS icon** copied from info tab.  

[svgbug_nov23.zip](https://github.com/godotengine/godot/files/10077660/svgbug_nov23.zip)

Admittedly, after pinpointing it to a Mac folder icon, not exactly wide use case...  
I'll just know that MacOS icons can't be coupled with PNGs in SVG format...

Thank you @akien-mga !!!

**Comment by metanoder** [comment]
Thank you for checking. Sorry if this was all a fire drill. 

Glad its not a problem. Maybe its my specific build with plugin...

**Comment by metanoder** [comment]
For posterity...

Downloaded Godot 4 beta 6.
Tired opening it in PC.  Ironically, it opens on PC, but not on Mac version of Godot...
